src/client.ts

```diff
@@ -600,6 +613,103 @@ export default class ConcordiumNodeClient {
         }
     }
 
+    /**
+     * Retrieve list of bakers on the network.
+     * @param blockHash the block hash to get the smart contact instances at
+     * @returns A JSON list of baker IDs
+     */
+    async getBakerList(blockHash: string): Promise<BakerId[] | undefined> {
+        if (!isValidHash(blockHash)) {
+            throw new Error('The input was not a valid hash: ' + blockHash);
+        }
+
+        const bh = new BlockHash();
+        bh.setBlockHash(blockHash);
+
+        const response = await this.sendRequest(this.client.getBakerList, bh);
+
+        return unwrapJsonResponse<BakerId[]>(
+            response,
+            (_, v) => BigInt(v as string),
+            intListToStringList
+        );
+    }
+
+    /**
+     * Gets the status the L-pool.
+     * @param blockHash the block hash the status at
+     * @returns The status of the L-pool.
+     */
+    async getPoolStatus(blockHash: string): Promise<LPoolStatus | undefined>;
+    /**
+     * Gets the status a baker.
+     * @param blockHash the block hash the status at
+     * @param bakerId the ID of the baker to get the status for.
+     * @returns The status of the corresponding baker pool.
+     */
+    async getPoolStatus(
+        blockHash: string,
+        bakerId: BakerId
+    ): Promise<BakerPoolStatus | undefined>;
+    /**
+     * Gets the status of either a baker, if a baker ID is supplied, or the L-pool if left undefined.
+     * @param blockHash the block hash the status at
+     * @param [bakerId] the ID of the baker to get the status for. If left undefined, the status of the L-pool is returned.
+     * @returns The status of the corresponding pool.
+     */
+    async getPoolStatus(
```

Comment:
Is this header necessary? :thinking:

Reply:
When you look at the resulting types, only the function type definitions, and NOT the implementation, is present. I needed the dynamic version of the function in addition to the others.